Code/GraphMol/RingInfo.cpp

```diff
@@ -16,77 +16,157 @@ RingInfo::INT_VECT RingInfo::atomRingSizes(unsigned int idx) const {
   PRECONDITION(df_init, "RingInfo not initialized");
 
   if (idx < d_atomMembers.size()) {
-    return d_atomMembers[idx];
-  } else {
-    return INT_VECT{0};
+    INT_VECT res(d_atomMembers[idx].size());
+    std::transform(d_atomMembers[idx].begin(), d_atomMembers[idx].end(),
+                   res.begin(),
+                   [this](int ri) { return d_atomRings.at(ri).size(); });
+    return res;
   }
+  return INT_VECT{0};
 }
 bool RingInfo::isAtomInRingOfSize(unsigned int idx, unsigned int size) const {
   PRECONDITION(df_init, "RingInfo not initialized");
 
   if (idx < d_atomMembers.size()) {
-    return std::find(d_atomMembers[idx].begin(), d_atomMembers[idx].end(),
-                     static_cast<int>(size)) != d_atomMembers[idx].end();
-  } else {
-    return false;
+    return std::find_if(d_atomMembers[idx].begin(), d_atomMembers[idx].end(),
+                        [this, size](int ri) {
+                          return d_atomRings.at(ri).size() == size;
+                        }) != d_atomMembers[idx].end();
   }
+  return false;
 }
 unsigned int RingInfo::minAtomRingSize(unsigned int idx) const {
   PRECONDITION(df_init, "RingInfo not initialized");
 
-  if (idx < d_atomMembers.size() && d_atomMembers[idx].size()) {
-    return *std::min_element(d_atomMembers[idx].begin(),
-                             d_atomMembers[idx].end());
-  } else {
-    return 0;
+  if (idx < d_atomMembers.size() && !d_atomMembers[idx].empty()) {
+    return d_atomRings
+        .at(*std::min_element(
+            d_atomMembers[idx].begin(), d_atomMembers[idx].end(),
+            [this](int ri1, int ri2) {
+              return d_atomRings.at(ri1).size() < d_atomRings.at(ri2).size();
+            }))
+        .size();
```

Comment:
this is a "a bit" too much for a one-liner.
how about:
```suggestion
   auto elem = *std::min_element(
            d_atomMembers[idx].begin(), d_atomMembers[idx].end(),
            [this](int ri1, int ri2) {
              return d_atomRings.at(ri1).size() < d_atomRings.at(ri2).size();
            });
    return d_atomRings.at(elem).size();
```
(needs to be clang-formatted)

Reply:
Done.